lib/Mir.ml

```diff
@@ -65,6 +65,7 @@ let rec remove_size = function
   | SArray (t, _) -> UArray (remove_size t)
 
 type litType = Int | Real | Str [@@deriving sexp, hash]
+type funKind = StanMath | UserDefined [@@deriving compare, sexp, hash]
```

Comment:
Unfortunately there are more options here than just StanMath or UserDefined... Maybe for now I'd put UserDefined or Builtin, assuming it's not too crazy to use the word builtin to refer to math library functions, internal mir functions, Stan functions, etc?